onboardingtaskapp.client/src/components/Product/List.jsx

```diff
@@ -9,59 +9,98 @@ export default function List() {
+                .then(data => setProducts(data))
+                .catch(error => console.error('Error fetching products:', error));
+
         } catch (error) {
-            console.error('There was a problem deleting product:', error.message);
+            console.error('There was a problem fetching product data:', error.message);
         }
+        
     };
 
+    const handleDeleteSuccess = async (isDeleted) => {
+
+        if (isDeleted) {
+            await fetchProducts();
+        }
+    }
+
+    const handleUpdateSuccess = async (isUpdated) => {
+
+        if (isUpdated) {
+            await fetchProducts();
+        }
+    }
+
+    const handleCreateSuccess = async (isCreated) => {
+
+        if (isCreated) {
+            await fetchProducts();
+        }
+    }
+
+    const renderTableRows = (product, handleUpdateSuccess, handleDeleteSuccess) => {
+
+        const productDeleteItem = {
+            id: product.id,
+            title: 'Delete Product',
+            buttonText: 'DELETE',
+            url: 'Product',
+            redirect: 'products'
+        };
+
+        return (
+            <tr key={product.id}>
+                <td data-label="Name">{product.name}</td>
+                <td data-label="Price">{product.price}</td>
+                <td>
+                    <ProductEdit
+                        item={{ product }}
+                        isUpdated={handleUpdateSuccess}
+                    />
+                </td>
+                <td>
+                    <ProductDelete
+                        item={ productDeleteItem }                       
+                        isDeleted={handleDeleteSuccess}
+                    />
+                </td>
+            </tr>
+
+        )
+    }
+
+
     return (
         <>
             
-            <ProductCreate />
+            <ProductCreate isCreated={handleCreateSuccess} />
 
             <table className="ui celled table">
                 <thead>
                     <tr><th>Name</th>
-                        <th>Address</th>
+                        <th>Price</th>
                         <th>Actions</th>
                         <th>Actions</th>
                     </tr>
                 </thead>
                 <tbody>
-                    {products.map(product => (
-                        <tr key={product.id}>
-                            <td data-label="Name">{product.name}</td>
-                            <td data-label="Address">{product.price}</td>
-                            <td>
-                                <ProductEdit item={{ product }} />                                
-                            </td>
-                            <td>
-                                <ProductDelete item={{ id: product.id, title: 'Delete Product', buttonText: 'DELETE', url: 'Product', redirect: 'products' }} />
-                            </td>
-                        </tr>
-                    ))}
+                    {products.map(product => 
```

Comment:
Please check my previous feedback about this kind of implementation, this is the same with my notes in the customer list. Thanks